main.go

```diff
@@ -114,7 +114,92 @@ func createKitty(ctx contractapi.TransactionContextInterface, matronID, sireID,
 	return newKittenID, nil
 }
 
-func (c *KittyContract) Transfer(ctx contractapi.TransactionContextInterface, from, to string, kittyID uint64) error {
+func owns(kittyID uint64, owner string) (bool, error) {
+	return false, nil
+}
+
+func approvedFor(kittyId uint64, account string) (bool, error) {
+	return false, nil
+}
+
+func approve(kittdyId uint64, account string) error {
+	return nil
+}
+
+func (c *KittyContract) Transfer(ctx contractapi.TransactionContextInterface, to string, kittyID uint64) error {
+	return nil
+}
+
+func (c *KittyContract) Approve(ctx contractapi.TransactionContextInterface, kittdyId uint64, account string) error {
+	return nil
+}
+
+func (c *KittyContract) TotalSuppy(ctx contractapi.TransactionContextInterface) uint64 {
+	return 0
+}
+
+func (c *KittyContract) OwnerOf(ctx contractapi.TransactionContextInterface, kittyId uint64) (string, error) {
+	return "", nil
+}
+
+func (c *KittyContract) TokensOfOwner(owner string) ([]uint64, error) {
+	return []uint64{}, nil
+}
+
+func (c *KittyContract) PregnantKitties() (uint64, error) {
+	return 0, nil
+}
+
+func isReadyToGiveBirth(matron Kitty) (bool, error) {
+	return true, nil
+}
+
+func isReadyToBreed(kittyId uint64) (bool, error) {
+	return false, nil
+}
+
+func (c *KittyContract) IsReadyToBreed(kittyId uint64) (bool, error) {
+	return false, nil
+}
+
+func isSiringPermitted(kittyId uint64) (bool, error) {
+	return true, nil
+}
+
+func triggerCooldown(kittyId uint64) error {
+	return nil
+}
+
+func (c *KittyContract) ApproveSiring(ctx contractapi.TransactionContextInterface, kittyId uint64, siringPartner string) error {
+	return nil
+}
+
+func isPregnant(kittyId uint64) (bool, error) {
+	return false, nil
+}
+
+func isValidMatingPair(sire uint64, matron uint64) (bool, error) {
+	return true, nil
+}
+
+func canBreedWith(sire uint64, matron uint64) (bool, error) {
```

Comment:
canBreedWith is also an external function in the Solidity implementation. I think this would make sense to have it here as well, so that a user can check if two kitties are a valid mating pair from the outside.